pkg/prometheus/common.go

```diff
@@ -594,3 +598,31 @@ func MakeProbes(
 
 	return startupProbe, readinessProbe, livenessProbe
 }
+
+// If a ServiceName is specified, check that the service exists in the namespace. If it does not exist, fail
+// the reconciliation.
+// Also, ensure that the Prometheus instance is selected by the service. If not, fail the reconciliation.
+func CheckCustomService(ctx context.Context, serviceName *string, prometheusNS string, prometheusName string, svcClient clientv1.ServiceInterface, selectorLabels map[string]string) error {
```

Comment:
What if we pass a PrometheusInterface parameter instead? It would simplify the function signature.

Reply:
I think at the time of writing, my POV was that do we need to pass the entire interface when we only need 3 string values from it. But no strong opinions on this one.